### PR TITLE
Add explanation about switching the three modes.

### LINKED ISSUE #5277: Document modes of BatchNormalization
Improve documentation on each modes of batch normzalization (train, finetune, test).
Context: #4553

### PR DESCRIPTION
**Pull request by fiarabbit**
fix #5277

### CODE REVIEW

**Comment by kmaehashi** [comment]
Jenkins, test this please.

File: chainer/links/normalization/batch_normalization.py [hunk 1]
```diff
@@ -21,17 +21,21 @@ class BatchNormalization(link.Link):
...
+    be used for instant evaluation in testing mode. Training mode is enabled
+    when ``chainer.config.train`` is set to ``True`` and :meth:`__call__`
+    is invoked with ``finetune=False`` (the default is False).
 
     In fine-tuning mode, it accumulates the input to compute *population
     statistics*. In order to correctly compute the population statistics, a
     user must use this mode to feed mini-batches running through whole training
-    dataset.
+    dataset. Finetuning mode is enabled when ``chainer.config.train`` is set to
+    ``True`` and :meth:`__call___` is invoked with ``finetune=True``.
```
Comment (kmaehashi) [inline chainer/links/normalization/batch_normalization.py]: `s/__call___/__call__/`

**Comment by hvy** [comment]
Could not backport before release so I removed the label.
